### DISCUSSION TITLE
math in markdown failed rendering

### DISCUSSION THREAD

**Discussion by jakub-homola** [community/community#41087] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

Hello,

(sorry if I am posting this to the wrong place, I could not find anything more appropriate)

I was writing some math in Markdown, and discovered, that this equation does not render (btw this is integration per partes):

`$\int_{a}^{b} u v^\prime \ \mathrm{d}x = [ uv ]_a^b - \int_{a}^{b} u^\prime v \ \mathrm{d}x$`

$\int_{a}^{b} u v^\prime \ \mathrm{d}x = [ uv ]_a^b - \int_{a}^{b} u^\prime v \ \mathrm{d}x$

I pinned the problem down to this, and have no idea what is wrong (the first three expressions render ok, the last one fails to render):

`$x_2 + \int_5$`

$x_2 + \int_5$

`$x_2 + \int_{5}$`

$x_2 + \int_{5}$

`$[x]_2 + \int_5$`

$[x]_2 + \int_5$

`$[x]_2 + \int_{5}$`

$[x]_2 + \int_{5}$

But when written separately, both sub-expressions are rendered correctly on their own:

`$[x]_2$`

$[x]_2$

`$\int_{5}$`

$\int_{5}$

Am I misunderstanding something, or is this a bug?

BTW, in mathjax this renders without problems, according to this: <https://saxarona.github.io/mathjax-viewer/>

Thanks for any response,

Jakub

**Comment by CompetitiveLin** [comment]
similar problem like i met:
$$true \&\& false$$

**Comment by jfullstackdev** [comment]
from the pattern I'm seeing 

![image](https://user-images.githubusercontent.com/47092464/206866135-479c4809-4e1d-4725-9ab0-3defed5078a5.png)

these are all working

when you have the last one

![image](https://user-images.githubusercontent.com/47092464/206866161-1d4b185d-1ec2-42bc-9907-3daf3e3ac752.png)

you both have `[]` and `{}`

but if you will be separating them, they will work

`$[x]_2$`

$[x]_2$

`$\int_{5}$`

$\int_{5}$

so, I think that is what you are referring and please do note, Markdown will not have same implementation because it's not standardized

**Reply by jakub-homola** [reply]
Yes, this is what I am refering to. I understand this is not standardized, and is probably just an issue here on github, where the math renderer and markdown renderer combined produce wrong results in this case.

I updated the OP to highlight that the separate expressions render well, but combined they do not.

**Reply by DSDmark** [reply]
$\int_{a}^{b} u v^\prime \ \mathrm{d}x = [ uv ]a^b - \int{a}^{b} u^\prime v \ \mathrm{d}x$

```$\int_{a}^{b} u v^\prime \ \mathrm{d}x = [ uv ]a^b - \int{a}^{b} u^\prime v \ \mathrm{d}x$```

**Reply by DSDmark** [reply]
$[x]2 + \int{5}$ 

```$[x]2 + \int{5}$```

**Reply by DSDmark** [reply]
:no_mouth:

**Comment by nschloe** [comment]
This is definitely a bug. I've condensed it down to
````markdown
$[x]_y a_{b}$
````
$[x]_y a_{b}$

Screenshot (Dec 15, 2022):

![screenshot](https://user-images.githubusercontent.com/181628/207810543-83382501-de04-4ec0-95b9-249989bcc148.png)

The bug is perhaps related to GitHub's poor separation between their Markdown sanitizer and inline math. The problem is that the sanitizer doesn't "know" $...$-math from any other text, and "fixes" things in it. GitHub have implemented certain rules to avoid that, but they don't cover all cases. A more consistent approach would be to enclose math in backticks, like inline code, e.g.,
````markdown
$`[x]_y a_{b}`$
````
which would avoid sanitizing altogether.

**Comment by nschloe** [comment]
I just noticed this is probably a duplicate of https://github.com/orgs/community/discussions/36915.

Added this to my bug list https://github.com/nschloe/github-math-bugs.

**Comment by nschloe** [comment, marked as answer]
This is fixed by the new backtick-delimited inline math syntax:

https://github.blog/changelog/2023-05-08-new-delimiter-syntax-for-inline-mathematical-expressions/

> $`\int_{a}^{b} u v^\prime \ \mathrm{d}x = [ uv ]_a^b - \int_{a}^{b} u^\prime v \ \mathrm{d}x`$

**Reply by jakub-homola** [reply]
Awesome, thanks! Is the dollar-backtick syntax more widely used or does it work just here on github?

$`[x]_2 + \int_{5}`$

$`t+e+s+t`$

``$`t+e+s+t`$``

**Reply by nschloe** [reply]
It's also GitLab's syntax choice (see [here](https://docs.gitlab.com/ee/user/markdown.html#math)). The reason for the backticks is to make sure the math contents behave well with the surrounding Markdown. To Markdown, everything in backticks is "code" and it won't touch it. Pure `$...$` means nothing to Markdown so it messes with the content, leading to many math bugs such as this one here.
